business/dashboards.go

```diff
@@ -562,16 +562,19 @@ func (in *DashboardsService) GetCustomDashboardRefs(namespace, app, version stri
 
 	if len(runtimes) == 0 {
 		cfg := config.Get()
+		discoveredRuntimes := map[string]models.Runtime{}
 		discoveryEnabled := cfg.ExternalServices.CustomDashboards.DiscoveryEnabled
 		if discoveryEnabled == config.DashboardsDiscoveryEnabled ||
 			(discoveryEnabled == config.DashboardsDiscoveryAuto &&
 				len(pods) <= cfg.ExternalServices.CustomDashboards.DiscoveryAutoThreshold) {
-			filters := make(map[string]string)
-			filters[cfg.IstioLabels.AppLabelName] = app
-			if version != "" {
-				filters[cfg.IstioLabels.VersionLabelName] = version
+			for _, appVersionLabelSelector := range cfg.GetAppVersionLabelSelectors(app, version) {
+				for _, discoveredDashboard := range in.discoverDashboards(namespace, appVersionLabelSelector.Requirements) {
+					discoveredRuntimes[discoveredDashboard.Name] = discoveredDashboard
```

Comment:
Won't the discoveredDashboard.Name be the same for each iteration in the loop, thus overwriting the one in prior iterations? If so, don't we lose data and the "for" loop below on line 575 pointless (since there will always be just one item in the array)??